pkg/sources/docker/docker.go

```diff
@@ -194,6 +207,26 @@ func (s *Source) processLayer(ctx context.Context, layer v1.Layer, imgInfo image
 		return err
 	}
 
+	// If using layer caching, check if layer already scanned. If no secrets found before, skip.
+	if s.conn.Cache {
+		db, err := ConnectToLayersDB(s.conn.CacheDb)
```

Comment:
suggestion: As mentioned in my previous comment, establishing and tearing down database connections for each Docker layer could impact performance. We should consider establishing connections earlier in the scanning process and utilize connection pooling to reduce the overhead and improve efficiency. This would align with the approach discussed earlier for optimizing database connections.